rs/state_manager/tests/state_manager.rs

```diff
@@ -3415,6 +3415,238 @@ fn can_recover_from_corruption_on_state_sync() {
     });
 }
 
+#[test]
```

Comment:
You mean where it would get stuck if we don't have the code to archive unverified checkpoint, right?

It will be `load_checkpoint_parallel` in the `StateManagerImp::new()`, i.e. the `restart_fn` function in the test.